main.py

```diff
@@ -1,132 +1,379 @@
 
 
-import argparse
-from sklearn_extra.cluster import KMedoids
+import os.path
+# import time
+
+import numpy
 import pandas as pd
 import sys
-DATABUFFER = 268435456 # size of quarter of a gigabyte
+import psutil
+import math
+from sklearn_extra.cluster import KMedoids
+# from sklearn.compose import ColumnTransformer
+# from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import LabelEncoder
 
-# def decide_row_num_halt(path,cols=None):
-#     if path[-5:] == ".xlsx" or path[-4:] == ".xls":
-#         temp = pd.read_excel(path,usecols=cols,nrows=1)
-#     elif path[-4:] == ".csv":
-#         temp=pd.read_csv(path,usecols=cols,nrows=1)
-#     else:
-#         return
-#     row_num= DATABUFFER/ sys.getsizeof(temp)
 
+def load_data_xls(path, startRows = None,
+                  endRows = None, cols = None, 
+                  sheet = 0):
+    header = pd.read_excel(path, usecols = cols, nrows = 1, sheet_name = sheet)
+    temp = pd.read_excel(path,
```

Comment:
Huh? Why do you access the file twice? Not sure I understand what's going on here

Reply:
if i dont take the headers (the first line, the titles) then the second read_excel doesnt take the column headers into account and i get columns with title "75.12". i later use the column names